### ISSUE TITLE
drag-and-drop fix for REAPER doesn't work with "Insert virtual instrument on new track..."

### ISSUE THREAD

**Issue by mrbumpy409** [robbert-vdh/yabridge#160]
### Thanks for giving yabridge a shot!

- [X] I read through both the troubleshooting and the known issues sections, and my issue wasn't listed there

### Problem description

1. Make sure you have the yabridge drag and drop fix for REAPER correctly set up via "yabridge.toml" files.
2. In REAPER, create a virtual instrument on a new track via "Track" -> "Insert virtual instrument on new track...", and select a plugin that you have configured for yabridge drag and drop support.
3. Drag and drop a file into the plugin.

### What did you expect to happen?

I expected the drag and drop operation to succeed.

### What actually happened?

REAPER throws the error "Could not import 1 of 1 items!"

### Operating system

KDE neon 5.23

### How did you install yabridge?

From the releases page, extracted to ~/.local/share as per the instructions.

### yabridge version

3.8.0

### yabridgectl version

3.8.0

### Wine version

Wine Staging 7.0

### Plugin

Plogue Sforzando, juicysfplugin

### Plugin type

both VST2 and VST3

### Plugin architecture

both 64-bit and 32-bit

### Host

REAPER v6.46/linux-x86_64 rev ae05ce (Jan 19 2022)

### Debug log

[yabridge.log](https://github.com/robbert-vdh/yabridge/files/7960524/yabridge.log)

### Anything else?

There are two workarounds to this issue:

1. Click the UI button twice to disable and re-enable the plugin UI.
2. Double-click the plugin name on left side of the track FX list to pop the UI out into its own floating window, then close this floating window to re-embed the plugin into the track FX list.

Now drag and drop will function correctly.

Adding a virtual instrument to an existing track via the FX list "Add" button does not trigger this bug.

**Comment by robbert-vdh** [comment]
Thanks a lot for the report! REAPER probably handles the window embedding a bit differently when you insert a plugin this way, I'll look into it in a bit.

**Comment by robbert-vdh** [comment]
REAPER does a weird thing where they create a window, have the plugin reparent itself to that window, and they then reparent their own window to another window. I've ran into this with them before because they do something similar when you reopen a closed editor window (https://github.com/robbert-vdh/yabridge/blob/d6260c1d6beb7f1d0669a61c8aa8325b891a985f/src/wine-host/editor.cpp#L473-L483). I've moved the DnD 'fixing' logic to the same place as where we redetect the current host window. That fixes the issue.

If you want to give it a try, then check out this section of the readme for instructions on how to install a yabridge development build: https://github.com/robbert-vdh/yabridge#installing-a-development-build
